experimental/learn_descriptors/four_seasons_parser.cc

```diff
@@ -0,0 +1,277 @@
+#include "experimental/learn_descriptors/four_seasons_parser.hh"
+
+#include <limits.h>
+
+#include <cmath>
+#include <exception>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <typeinfo>
+#include <unordered_map>
+#include <utility>
+
+enum class GPSIdx {
+    TIME_NS = 0,
+    TRAN_X = 1,
+    TRAN_Y = 2,
+    TRAN_Z = 3,
+    QUAT_X = 4,
+    QUAT_Y = 5,
+    QUAT_Z = 6,
+    QUAT_W = 7,
+};
+enum class ImgIdx {
+    TIME_NS = 0,
+    TIME_SEC = 1,
+    EXPOSURE_TIME = 2  // I'm not completely sure if this is correct - Nico
+};
+enum class ResultIdx {
+    TIME_SEC = 0,
+    TRAN_X = 1,
+    TRAN_Y = 2,
+    TRAN_Z = 3,
+    QUAT_X = 4,
+    QUAT_Y = 5,
+    QUAT_Z = 6,
+    QUAT_W = 7,
+};
+enum class CalibIdx { FX = 1, FY = 2, CX = 3, CY = 4, K1 = 5, K2 = 6, K3 = 7, K4 = 8 };
+
+std::vector<std::string> parse_line_adv(const std::string& line, const std::string& delim = " ") {
```

Comment:
Might this  function be easier to write if you used of of abseil's [String Utilities](https://abseil.io/docs/cpp/guides/strings)?